presto-spi/src/main/java/com/facebook/presto/spi/type/SqlIpAddress.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.type;
+
+import java.util.Arrays;
+
+import static com.facebook.presto.spi.type.IpAddressType.IP_ADDRESS_SIZE;
+import static java.lang.System.arraycopy;
+import static java.util.Objects.requireNonNull;
+
+public class SqlIpAddress
```

Comment:
This should go in `presto-main`, not in the SPI. Normal types don't need to be in the SPI. (Ideally, all type implementations would be in presto-main)